crates/sui-types/src/temporary_store.rs

```diff
@@ -958,8 +950,8 @@ impl<S: ObjectStore> TemporaryStore<S> {
     ) -> GasCostSummary {
         // at this point, we have done *all* charging for computation,
         // but have not yet set the storage rebate or storage gas units
-        assert!(gas_status.storage_rebate() == 0);
-        assert!(gas_status.storage_gas_units() == 0);
+        debug_assert!(gas_status.storage_rebate() == 0);
+        debug_assert!(gas_status.storage_gas_units() == 0);
```

Comment:
@sblackshear not sure if we should change those or not. 
Those asserts were completely harmless and we had them there more for documentation than anything else. At the time we introduced them this was only called after execution.
Later we have added a re-entrancy from conservation checks failure and that is causing the assert to fire.
I am going to look in more details how to be more correct but wonder what people think about turning them off.
or alternatives

Reply:
I agree with turning these into `debug_asserts`--IMO `assert`s should be reserved for cases where the alternative is known to be so bad that crashing is worth it. And we've seen hitting this `assert` stopping us from hitting a more understandable `invariant_violation` a few times now.

Reply:
protected the change with a version check